src/components/PeopleFilters.tsx

```diff
@@ -1,18 +1,62 @@
+import { useSearchParams } from 'react-router-dom';
+import { SearchParams, getSearchWith } from '../utils/searchHelper';
+import { SearchLink } from './SearchLink';
+import classNames from 'classnames';
+
 export const PeopleFilters = () => {
+  const [searchParams, setSearchParams] = useSearchParams();
+  const query = searchParams.get('query') || '';
+  const sex = searchParams.get('sex') || '';
+  const centuries = searchParams.getAll('centuries') || [];
+
+  const updateQueryParams = (params: SearchParams) => {
+    const search = getSearchWith(searchParams, params);
+
+    setSearchParams(search);
+  };
+
+  const handleQueryChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    updateQueryParams({ query: event.target.value || null });
+  };
+
+  const toggleCenturyFilter = (century: string) => {
+    return centuries.includes(century)
+      ? centuries.filter(cent => cent !== century)
+      : [...centuries, century];
+  };
```

Comment:
The `toggleCenturyFilter` function is designed to toggle the presence of a century in the `centuries` array. However, when updating the `params` in the `SearchLink` component (line 88), it might not correctly reflect the toggled state because it directly uses the return value of `toggleCenturyFilter`. Consider storing the result in a variable before passing it to `params` to ensure the state is correctly updated.